src/node.ts

```diff
@@ -227,29 +238,41 @@ class Node extends VEventTarget {
       }
     });
     if (!this.resolved && this.resolvedInputs === this.inputCount) {
-      this.inputsReady();
+      // eslint-disable-next-line consistent-return
+      return this.inputsReady();
     }
   }
 
-  inputsReady(): void {
-    this.log('Node action:', this.name);
-    this.dumpInputs();
-    let p;
-    try {
-      p = this.action();
-    } catch (err) {
-      this.actionError(err);
-      return;
-    }
-    if (p instanceof Promise) {
-      p
-        .then(
-          () => { this.actionReady(); },
-          (err) => { this.actionError(err); },
-        );
-    } else {
-      this.actionReady();
-    }
+  inputsReady(): void | Promise<void> {
+    const promis = new Promise<void>((resolve, reject) => {
```

Comment:
- "promis" a very generic variable name. In this case, perhaps just `return new Promise(...)`?;
- "resolve" - confusing clash with this.resolve()